### PR TITLE
Update testing framework

### PR DESCRIPTION
**Pull request by eherot**
# BREAKING

Just a note that the removal of `storage::ec2test` is breaking.

### CODE REVIEW

File: .kitchen.yml [hunk 1]
```diff
@@ -1,39 +1,31 @@
...
-      Env: public
 - name: ubuntu-14.04
-  driver:
-    image_id: ami-d0ba0cb8
-    tags:
-      Name: storage-default-ubuntu-1404
-      Env: public
+  tags:
+    Name: storage-default-ubuntu-1404
+    Type: test
```
Comment (jeffbyrnes) [inline .kitchen.yml]: Wait, I thought it needs to be `driver.tags`? Also, I’d be in favor of this:

```yaml
driver:
  tags:
    Type: test
    Operator: <%= ENV['USER'] %>

suites:
  - name: ec2
    driver:
      tags:
        Name: storage-default-ubuntu-1404
```

I think it is a good idea to always attach the `tags.Name` to a particular suite, even in cases where we only have one. Consistent with the few cases where we have more than one suite.

**Comment by jeffbyrnes** [comment]
One bit about the `driver.tags`, and some thoughts on a more consistent TK config.

**Comment by eherot** [comment]
Updated the tags stuff.

**Comment by jeffbyrnes** [comment]
Looks good, seems like you’d like to discuss my other TK config thoughts some more (since they’re not there). This is good to merge in as-is, other changes can wait.

**Comment by eherot** [comment]
What other comments are you talking about?

File: .kitchen.yml [hunk 2]
```diff
@@ -1,39 +1,33 @@
...
-    image_id: ami-d0ba0cb8
-    tags:
-      Name: storage-default-ubuntu-1404
-      Env: public
 
 suites:
   - name: ec2
+    driver:
+      tags:
+        Name: storage-ec2-ubuntu-1404
```
Comment (eherot) [inline .kitchen.yml]: Updated this name to match the suite name.

**Comment by jeffbyrnes** [comment]
I refer to my suggestion of where to place `driver.tags.Type`, and the inclusion of the `Operator` tag. I didn’t do a great job, so let me elaborate:

For any tags that apply to all suites, I’d propose we always place then at the top level, under the `driver.tags` keyspace. This should always include `Type: test` and `Operator: <%= ENV['USER'] %>`, as these are valuable & easy.

For tags that only apply to a particular suite, which is usually just `Name`, they should be placed under the suite in question, which you have already done here, and we typically do on multi-suite test cases.

I’d propose that we _always_ do that, even in cases where we only have a single suite, for reasons of consistency.

**Comment by eherot** [comment]
Oh I see what you mean. That probably makes sense. Let me update.

**Comment by eherot** [comment]
try now.

**Comment by jeffbyrnes** [comment]
LGTM :shipit:

**Pull request merged**